scripts/covid_hosp_explore.R

```diff
@@ -1,37 +1,37 @@
 source("scripts/targets-common.R")
 source("scripts/targets-exploration-common.R")
 
+# ================================ GLOBALS =================================
 # These globals are needed by make_forecasts_and_scores (and they need to persist
 # during the actual targets run, since the commands are frozen as expressions).
-hhs_signal <- "confirmed_admissions_covid_1d"
-if (!exists("ref_time_values_")) {
-  # Alternatively you can let slide_forecaster figure out ref_time_values
-  start_date <- as.Date("2023-11-08")
-  end_date <- as.Date("2024-04-24")
-  # end_date <- start_date + 7
-  date_step <- 7L
-  ref_time_values_ <- seq.Date(start_date, end_date, by = date_step)
-}
-time_value_adjust <- 3 # this moves the week marker from Saturday to Wednesday
-
-# Debug mode will replace all forecasters with a fast dummy forecaster. Helps
-# with prototyping the pipeline.
-dummy_mode <- as.logical(Sys.getenv("DUMMY_MODE", FALSE))
-
-# these are for datasets which have some locations with extreme latency; currently that's only nwss
-# note that this has a vector of names first, and then the list of things to
-# exclude, because targets hates named lists and strips their names
-very_latent_locations <- list(list(
-  c("geo_value"),
-  c("la", "ms", "nd")
-))
-# TODO decide on this, nwss is latent in these locations by ~56 days. might be ok
+config <- list(
+  aheads = 0:4 * 7,
+  hhs_signal = "confirmed_admissions_covid_1d",
+  # The date when the forecast was generated (this is effectively the AS OF date).
+  forecast_generation_dates = seq.Date(as.Date("2023-11-08"), as.Date("2024-04-24"), by = 7L),
+  # The reference date for the forecast.
+  forecast_dates = seq.Date(as.Date("2023-11-08"), as.Date("2024-04-24"), by = 7L),
+  # This moves the week marker from Saturday to Wednesday
+  time_value_adjust = 3,
+  # Directory for reports.
+  reports_dir = "reports",
+  # Fetch arguments for epidatr.
+  fetch_args = epidatr::fetch_args_list(return_empty = FALSE, timeout_seconds = 400),
+  # Debug mode will replace all forecaster functions with a fast dummy forecaster. Helps
+  # with prototyping the pipeline.
+  dummy_mode = as.logical(Sys.getenv("DUMMY_MODE", FALSE)),
+  insufficient_data_geos = c("as", "pr", "vi", "gu", "mp")
+)
+# For testing, reduce dates
+# config$forecast_generation_dates <- config$forecast_generation_dates[1:3]
+# config$forecast_dates <- config$forecast_dates[1:3]
 
+# ================================ FORECASTER PARAMETERS ====================
 # Human-readable object to be used for inspecting the forecasters in the pipeline.
-forecaster_parameter_combinations_ <- rlang::list2(
+config$forecaster_parameter_combinations <- rlang::list2(
```

Comment:
seems like overkill to stick an object this complicated into the config object

Reply:
Agree now, though collecting and explicitly marking our large bundle of global variables still seems like a good intention. The bigger problem now is that any change to `config$forecast_parameter_combinations` will invalidate all targets that use config at all... Still want to find a "best of" solution here